bitcoin/src/blockdata/transaction.rs

```diff
@@ -937,7 +927,7 @@ pub const fn predict_weight_from_slices(
     let mut i = 0;
     while i < inputs.len() {
         let prediction = inputs[i];
-        partial_input_weight += prediction.weight().to_wu() as usize;
+        partial_input_weight += prediction.witness_weight().to_wu() as usize;
```

Comment:
@yancyribbens @apoelstra AFAICT this should also be updated to use `total_weight()`. As it stands it's passing a partial weight to `predict_weight_internal` which now expects the total weight...

Reply:
Hmm that's annoying.  Thanks for catching this.  There must not be a test-case for this that failed now that the semantics of `predict_weight_internal` has changed.

Reply:
Yeah, great catch. @yancyribbens can you add a unit test and a fix in a new PR?

I think the code before these renames was basically unreadable so it would not surprise me to find bugs like this.

Reply:
>  @yancyribbens can you add a unit test and a fix in a new PR?

Yeah I can add it to PR: https://github.com/rust-bitcoin/rust-bitcoin/pull/3663